### PR TITLE
Role Privilege Datasource Implementation

### PR DESCRIPTION
**Pull request by RayLiu7**
# Description
Role Privilege Datasource Implementation
Role Privilege Datasource allows our customer to find available role privileges while creating/updating the roles.

# ISSUE TYPE
- Feature Pull Request

##### DATASOURCE NAME
Role Privilege

##### OUTPUT
image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

File: docs/data-sources/roleprivilege.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+# Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (NavisJ) [inline docs/data-sources/roleprivilege.md:2]: Copyright year should be 2024

**Review by NavisJ — APPROVED** [review]

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

**Comment by forrestxia** [comment]
run e2e test

File: docs/data-sources/roleprivilege.md (hunk 1 above)
Comment (RayLiu7) [reply docs/data-sources/roleprivilege.md:2]: @NavisJ I changed the copyright year of new go files to 2024, but changing the copyright year of doc file will need to modify the doc template. Directly change the doc will fail the "Terraform-main-CI / Generate". We can note it down for now and change it later for all new features.

**Comment by RayLiu7** [comment]
run e2e test

**Pull request merged**